galaxy_importer/collection.py

```diff
@@ -35,20 +35,81 @@
 CollectionFilename = namedtuple("CollectionFilename", ["namespace", "name", "version"])
 
 
-def import_collection(file, filename=None, file_url=None, logger=None, cfg=None):
+def import_collection(
+    file=None, filename=None, file_url=None, git_clone_path=None, logger=None, cfg=None,
+):
     """Process import on collection artifact file object.
 
+    If `git_clone_path` is provided, return metadata and file,
+    else return metadata.
+
     :raises exc.ImporterError: On errors that fail the import process.
     """
     logger.info(f"Importing with galaxy-importer {__version__}")
     if not cfg:
         config_data = config.ConfigFile.load()
         cfg = config.Config(config_data=config_data)
     logger = logger or default_logger
+
+    if (file and git_clone_path) or not (file or git_clone_path):
+        raise exc.ImporterError(
+            "Expected either 'file' or 'git_clone_path' to be populated"
+        )
+
+    if git_clone_path:
+        # TODO: use cfg.tmp_root_dir plus subdir?
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            file, filename = _build_collection(git_clone_path, empty_output_path=tmp_dir, logger=logger)
+            logger.info(f"file={file}")
+            metadata = _import_collection(file, filename, file_url, logger, cfg)
+            return (metadata, file)
+
     return _import_collection(file, filename, file_url, logger, cfg)
 
 
+def sync_collection(git_clone_path=None, logger=None, cfg=None):
```

Comment:
In what case would I want to call `sync_collection` vs. `import_collection`? It seems like they do the same thing when a `git_clone_path` is provided.

Reply:
This method needs to accept a path to where the built collection should be stored (--output-path).

Reply:
when an end user is publishing/importing to galaxy_ng they will expect the collection to be linted, in this case I assumed pulp-ansible will continue to call `import_collection` whether or not a gitref is used.

in a separate use case, during sync, we never want linting to occur since that can take a several minutes per collection version depending on the different `ansible-test sanity` tests that are run, so I figured it would be helpful to have a separate `sync_collection` method to call for that.